packages/react-components/src/components/trace-context-component.tsx

```diff
@@ -20,6 +20,7 @@ import { XYOutputComponent } from './xy-output-component';
 import { NullOutputComponent } from './null-output-component';
 import { AbstractOutputProps } from './abstract-output-component';
 import * as Messages from '@trace-viewer/base/lib/message-manager';
+import { ThemeService } from '@theia/core/lib/browser/theming';
```

Comment:
"react-component" and "base" are supposed to be independent from Theia (and VsCode). So, the code to detect the theme and update of theme needs to be under viewer-prototype which is Theia specific. From there you need to send a signal that the trace-context-compenent is receiving and updating according to the change. 

See for the signals in base/signal-manager.ts and how they are used.

Reply:
Thank you for the suggestion. I moved the logic to trace-viewer.